### PR TITLE
Add table name to error message for transactional operations

### PR DESCRIPTION
**Pull request by rchilaka-amzn**
The current error messaging does not include table names when an exception is encountered during a transaction. This change provides a sequential list of all tables on which a transactional operation was attempted. Please note that the purpose of including table names is to enable human debugging without causing confusing (like in https://github.com/pynamodb/PynamoDB/issues/816).

### CODE REVIEW

File: pynamodb/connection/base.py [hunk 1]
```diff
@@ -427,11 +427,19 @@ def _make_api_call(self, operation_name, operation_kwargs):
...
                     # Batch operations can hit multiple tables, report them comma separated
-                    verbose_properties['table_name'] = ','.join(operation_kwargs['RequestItems'])
+                    verbose_properties['table_name'] = ','.join(operation_kwargs[REQUEST_ITEMS])
+                elif TRANSACT_ITEMS in operation_kwargs:
+                    # Transactional operations can also hit multiple tables, or have multiple updates within
+                    # the same table
+                    items = []
+                    for item in operation_kwargs[TRANSACT_ITEMS]:
+                        for op in item.keys():
+                            items.append(item[op][TABLE_NAME])
```
Comment (ikonst) [inline pynamodb/connection/base.py:439]:
```suggestion
                    table_names = []
                    for item in operation_kwargs[TRANSACT_ITEMS]:
                        for op in item.values():
                            table_names.append(op[TABLE_NAME])
```

File: pynamodb/connection/base.py [hunk 2]
```diff
@@ -427,11 +427,19 @@ def _make_api_call(self, operation_name, operation_kwargs):
...
-                if 'RequestItems' in operation_kwargs:
+                if REQUEST_ITEMS in operation_kwargs:
                     # Batch operations can hit multiple tables, report them comma separated
-                    verbose_properties['table_name'] = ','.join(operation_kwargs['RequestItems'])
+                    verbose_properties['table_name'] = ','.join(operation_kwargs[REQUEST_ITEMS])
+                elif TRANSACT_ITEMS in operation_kwargs:
+                    # Transactional operations can also hit multiple tables, or have multiple updates within
+                    # the same table
+                    table_names = []
+                    for item in operation_kwargs[TRANSACT_ITEMS]:
```
Comment (garrettheel) [inline pynamodb/connection/base.py:437]: Should we de-duplicate table names here, or is there a desire to see them ordered corresponding to the items in the transaction?

Comment (garrettheel) [reply pynamodb/connection/base.py:437]: This seems okay. A little redundant when working with a single table, but I can definitely see it being useful for multi-table

**Review by garrettheel — APPROVED** [review]
Thank you!

**Pull request merged**